src/app/components/crypto-asset-item/crypto-asset-item.layout.tsx

```diff
@@ -33,39 +36,50 @@ export function CryptoAssetItemLayout({
   onSelectAsset,
   titleLeft,
   titleRightBulletInfo,
+  renderRightElement,
 }: CryptoAssetItemLayoutProps) {
-  const { availableBalanceString, dataTestId, formattedBalance } =
-    parseCryptoAssetBalance(availableBalance);
+  let titleRight: ReactNode | undefined = undefined;
```

Comment:
I don't think this is a good approach, using `let` and updating `titleRight` and `captionRight` .

If you use `const` it can help improve the maintainability and design of the code and reduce the complexity.